sharedmem.c

```diff
@@ -23,7 +23,7 @@ struct rmtfs_mem {
 };
 
 static int parse_hex_sysattr(struct udev_device *dev, const char *name,
-			     unsigned long *value)
+			     unsigned long long *value)
```

Comment:
Strictly speaking, unsigned long long isn't identical to uint64_t, so there is still a possibility that the code runs into the same incompatible pointer type issue on a different platform.

Reply:
True, but it does fix any conceivable compatibility problem I'm aware of. I guess I could make this take a uint64_t pointer, and then explicitly assign the ULL to the pointer after validating it.